pilot/pkg/model/context.go

```diff
@@ -101,26 +100,14 @@ type Proxy struct {
 
 	// Metadata key-value pairs extending the Node identifier
 	Metadata map[string]string
-
-	// mutex control access to mutable fields in the Proxy. On-demand will modify the
-	// list of services based on calls from envoy.
-	mutex sync.RWMutex
-
-	// serviceDependencies, if set, controls the list of outbound listeners and routes
-	// for which the proxy will receive configurations. If nil, the proxy will get config
-	// for all visible services.
-	// The list will be populated either from explicit declarations or using 'on-demand'
-	// feature, before generation takes place. Each node may have a different list, based on
-	// the requests handled by envoy.
-	serviceDependencies []*Service
 }
 
 // NodeType decides the responsibility of the proxy serves in the mesh
 type NodeType string
 
 const (
-	// Sidecar type is used for sidecar proxies in the application containers
-	Sidecar NodeType = "sidecar"
+	// SidecarProxy type is used for sidecar proxies in the application containers
+	SidecarProxy NodeType = "sidecar"
```

Comment:
Why is the rename needed ?

Reply:
Because it conflicts with the model.Sidecar CRD type.

Reply:
Can we call the new type something else ( or do this rename in separate PR, it makes review harder) ?

Reply:
Never mind - too much wasted time to break this part and it's harmless ( just lots of changes in the PR ).